controllers/highlightController.js

```diff
@@ -0,0 +1,232 @@
+import joi from "joi";
+import models from "../models";
+import constants from "../helpers/constants";
+import errorMessages from "../helpers/errorMessages";
+import {
+  highlightValidator,
+  editHighlightValidator,
+  oneHighlightValidator
+} from "../helpers/highlightValidator";
+
+const { Highlight } = models;
+const {
+  OK,
+  INTERNAL_SERVER_ERROR,
+  BAD_REQUEST,
+  CREATED,
+  NOT_FOUND
+} = constants.statusCode;
+const { INTERNAL_SERVER_ERROR_MESSAGE } = errorMessages;
+
+/**
+ * @class HighlightController
+ */
+class HighlightController {
+  /**
+   * @description Highlight a given article
+   * @param  {Object} req - The request object.
+   * @param  {Object} res - The response object.
+   * @param  {Object} next - The next middleware.
+   * @return {Object} - It returns the response object.
+   */
+
+  static create(req, res, next) {
+    const { slug } = req.params;
+    const { start, end } = req.query;
+    const { comment } = req.body;
+    const { article } = req;
+    const { id: userId } = req.user;
+    // eslint-disable-next-line no-unused-vars
+    joi.validate({ start, end }, highlightValidator, async (err, _value) => {
```

Comment:
@espoirMur It is a best practise to separate concerns, that means a unit should only address a particular task or functionality. Try to separate the request validation from the actual logic. You can instead validate the request's body directly from the route(endpoint)
Here a resource about [SoC](https://medium.com/@ariel.salem1989/separation-of-concerns-essential-javascript-1e30994fa7a5)

Reply:
Nice feedback bro, You mean having a separate middleware for validation?

Reply:
@espoirMur Yes, using a middleware that handles the validation